test/tools/unit_test_runner.d

```diff
@@ -0,0 +1,326 @@
+                    {
+                        test.throwable = t;
+                        failedTests ~= test;
+                    }
+                }
+            }
+
+            printReport();
+
+            UnitTestResult result = {
+                runMain: false,
+                executed: testCount,
+                passed: testCount - failedTests.length
+            };
+
+            return result;
+        }
+    }.outdent;
+
+    const imports = moduleNames
+        .map!(e => format!"import %s;"(e))
+        .joiner("\n")
+        .to!string;
+
+    const modules = moduleNames
+        .map!(e => format!`"%s"`(e))
+        .joiner(",\n")
+        .to!string;
+
+    const content = format!codeTemplate(imports, modules, format!`"%s"`(filter));
+    write(path, content);
+}
+
+/**
+Writes a cmdfile with all the compiler flags to the given `path`.
+
+Params:
+    path = the path where to write the cmdfile file
+    runnerPath = the path of the unit test runner file outputted by `writeRunnerFile`
+    outputPath = the path where to place the compiled binary
+    testFiles = the test files to compile
+*/
+void writeCmdfile(string path, string runnerPath, string outputPath,
+    const string[] testFiles)
+{
+    enum staticFlags = [
+        "-version=NoBackend",
+        "-version=GC",
+        "-version=NoMain",
+        "-version=MARS",
+        "-unittest",
+        "-J" ~ generatedDir.buildPath("dub"),
+        "-J" ~ projectRootDir.buildPath("res"),
+        "-I" ~ projectRootDir.buildPath("src"),
+        "-I" ~ unitTestDir,
+        "-i",
+        "-g",
+        "-main"
+    ];
+
+    const commonFlags = staticFlags ~
+        testFiles ~
+        runnerPath ~
+        ("-of" ~ outputPath) ~
+        ("-m" ~ model);
+
+    const flags = needsStrtold ? commonFlags ~ ("-L" ~ strtoldObjPath) : commonFlags;
+
+    write(path, flags.join("\n"));
+}
+
+/**
+Returns `true` if any of the given files don't exist.
+
+Also prints an error message.
+*/
+bool missingTestFiles(const string[] givenFiles)
+{
+    const nonExistingTestFiles = givenFiles
+        .map!(file => testPath(file))
+        .filter!(file => !file.exists)
+        .join("\n");
+
+    if (!nonExistingTestFiles.empty)
+    {
+        stderr.writefln("The following test files don't exist:\n\n%s",
+            nonExistingTestFiles);
+
+        return true;
+    }
+
+    return false;
+}
+
+void execute(const string[] args ...)
+{
+    enforce(spawnProcess(args).wait() == 0,
+        "Failed to execute command: " ~ args.join(" "));
+}
+
+void runConfig()
+{
+    enum configPath = projectRootDir.buildPath("config.d");
+    enum outputPath = generatedDir.buildPath("dub");
+    enum versionPath = projectRootDir.buildPath("VERSION");
+
+    auto cmd = [
+        dmdPath,
+        "-m" ~ model,
+        "-run",
+        configPath,
+        outputPath,
+        versionPath
+    ];
+
+    version (Posix)
+        cmd ~= "/etc";
+
+    execute(cmd);
+}
```

Comment:
Why do we need this? This file will be generated as part of either building DMD or building DMD as a library with Dub.

Reply:
This is not using Dub anymore. I failed to get it to install. But it might not be required since the assumption is that DMD is built anyway. I'll see if I can remove it.